### ISSUE TITLE
C# port of NanoVG project ... no pointers, 100% managed

### ISSUE THREAD

**Issue by migueltk** [memononen/nanovg#332]
Hello and thank you for this fabulous project.

My apologies if not the right place to present my work. The work is done and just want to share it with more people.

I finished the port to the programming language C# "I do not get along very well with pointers", the source code is here "https://sites.google.com/site/bitiopia/" ...

The code is correct, it works fine (tested on Windows 7 with native platform OpenTK and virtualized Ubuntu with VirtualBox), but in Ubuntu (also in Fedora 24) there is a strange problem between (OpenTK and Gallium 0.4) that does not set the "stencil-buffer" correctly.

Any help will be appreciated.

regards

**Comment by dpethes** [comment]
Hi, thanks for the port, works fine on Windows 10 (OpenTK)! However it would be easier to find and work with if you submitted the code to Github (there are a couple of methods missing, I started adding them).

**Comment by migueltk** [comment]
Hi, 'dpethes', thank you for the recognition ...

I have no plans to publish the code in 'Github', this involves taking time to get to know 'Github', how it works, my English is basic (I have to use the translator constantly), and the project does not update it at a rate that justifies the use of 'Github' I think, etc ... but I encourage you to post it yourself ...

**Comment by cambiata** [comment]
Great work, Miguel!

Tried your port in Xamarin/Win10, and with some tiny minor it works well. Thanks!

While at it, I grabbed your code and created a repo for it:
https://github.com/cambiata/NanoVGDotNet

Best! / Jonas

**Comment by migueltk** [comment]
Hi, Jonas ...

Thank you for creating the repository. You can be sure that any changes, improvements or documentation that I make will let you know.
The source code "hides" a lot of knowledge and reveal that knowledge, to me, is even harder to port the source code, but I'm stubborn, I just need a long time.

Greetings ... Miguel

**Comment by jpbruyere** [comment]
there was a small bug in OpenTK < 2.0 for setting stencil buf on linux with sdl2 platform, it should be gone now.

**Comment by cambiata** [comment]
Hi guys!

I'm not a C# guy at all, mostly in it because my intrest and use of Haxe (haxe.org), wich transpiles to C#, and I'm constantly looking for graphical interfaces and stuff for desktop applications. (Working on solutions for music notation.)

I only briefly tried your port on my win machines - works well! - and on my Ubunty 16.04 - runs, but with graphical glitches. (Is that maybe what you are referring to just here above, @jpbruyere ?

Most likely my repo will be untouched so feel free to fork it and take ownership over it, anyone of you!

**Comment by migueltk** [comment]
@jpbruyere 
If you mean this problem, https://github.com/opentk/opentk/issues/444, unfortunately the problem continues ...

**Comment by jpbruyere** [comment]
@migueltk I develop on debian 8 with latest mesa from source and opentk-2.0 nuget pkg and stencil buffer is working well in my [Chess game](https://github.com/jpbruyere/Chess) (I use stencil for reflexion). I tested your example, and it's not working, so maybe there's a problem with GL calls or config.

**Comment by jpbruyere** [comment]
@migueltk After doing some more test, it appears that specifying explicitly GL major/minor version solve the stencil buf creation (`base(800, 600, gm, "test",GameWindowFlags.Default,DisplayDevice.Default,3,0,GraphicsContextFlags.Debug)`)

**Comment by migueltk** [comment]
@jpbruyere 
Yes, yes, great! Checked, it is necessary to specify the version number. I have tested from 2.0 to 3.0 (on my hardware) and it works correctly, thanks.
This failure forced me to work with GLFW but OpenTK has always been my first choice.

**Comment by bostich83** [comment]
@migueltk 
would it be possible to change your .net-port license to something similar like the original lib (zlib) or MIT?

**Comment by migueltk** [comment]
@bostich83 
I'm sorry, but I don't consider the license change.
